test/k8sT/Policies.go

```diff
@@ -1581,30 +1582,21 @@ var _ = Describe("K8sPolicyTest", func() {
 		}
 
 		testConnectivitytoRedis := func() {
-			webPods, err := kubectl.GetPodsNodes(helpers.DefaultNamespace, "-l k8s-app.guestbook=web")
-			Expect(err).To(BeNil(), "Cannot get web pods")
-
-			serviceIP, port, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, "redis-master")
-			Expect(err).To(BeNil(), "Cannot get hostPort of redis-master")
-
-			serviceName := "redis-master"
-			err = kubectl.WaitForKubeDNSEntry(serviceName, helpers.DefaultNamespace)
-			Expect(err).To(BeNil(), "DNS entry is not ready after timeout")
+			webPods, err := kubectl.GetPodsNodes(helpers.DefaultNamespace, "-l app=guestbook")
+			Expect(err).To(BeNil(), "Error retrieving web pods")
+			Expect(webPods).ShouldNot(BeEmpty(), "Cannot retrieve web pods")
 
+			cmd := helpers.CurlFailNoStats(`"127.0.0.1/guestbook.php?cmd=set&key=messages&value=Hello"`)
```

Comment:
I'm a little confused about this part. Previously we would resolve two things: A service IP + Port (which confusingly we seem to be referring to as a `hostPort` but I'm not entirely clear on whether that's actually the service type in use) + a DNS name (with the same port).

With this `127.0.0.1` access, are we even guaranteed to egress the pod & hit the BPF code to test that we handle the application correctly? I would have expected that even if we think that removing the DNS angle for this PR (eg to improve reliability I guess?), we should still use a service IP + port to resolve.

Reply:
Yeah, so we used to make TCP connections from the frontend pods to both the backend pod and the backend service. That isn't really testing the multi-tier application end-to-end and I think frontend pods are supposed to reach the backend via the service.

I spent some time playing with a GuestBook app. deployed on a GKE cluster to figure out the best way to test the same as an actual client (ex. me on WebUI) would. Turns out the frontend does AJAX calls to the "PHP backend" (same pod) which connects to the Redis backend. That's what the new curl request+json parsing simulates. I then checked that the same curl request+json parsing fails when a backend is disconnected. (Just a curl request isn't enough as a failure to connect the backend may produce a HTTP 200 :frowning_face:)

(That being said, I'm not sure what's the overall purpose of this test with regard to Cilium. Is it even testing anything that's not covered elsewhere?)